ZuluContent/Engines/Craft/Core/Fortify.cs

```diff
@@ -0,0 +1,149 @@
+                TargetFlags.None)
+            {
+                m_From = from;
+                m_CraftSystem = craftSystem;
+                m_Tool = tool;
+            }
+
+            protected override void OnTarget(Mobile from, object targeted)
+            {
+                int num = m_CraftSystem.CanCraft(from, m_Tool, null);
+
+                if (num > 0)
+                {
+                    from.SendGump(new CraftGump(from, m_CraftSystem, m_Tool, num));
+                    return;
+                }
+
+                if (targeted is BaseHat hat)
+                {
+                    if (!hat.IsChildOf(from.Backpack))
+                    {
+                        from.SendFailureMessage("The hat has to be in your backpack.");
+                        return;
+                    }
+
+                    if (hat.Fortified == HatFortification.Fortified)
+                    {
+                        from.SendFailureMessage("That is fortified already!");
+                        return;
+                    }
+
+                    if (hat.HitPoints != hat.MaxHitPoints)
+                    {
+                        from.SendFailureMessage("You must repair that first.");
+                        return;
+                    }
+
+                    from.Target = new InternalTarget2(m_From, m_CraftSystem, m_Tool, hat);
+                    from.SendSuccessMessage("Select the helm to fortify this hat with.");
+                }
+                else
+                {
+                    from.SendFailureMessage("That's not a hat.");
+                }
+            }
+        }
+
+        private class InternalTarget2 : Target
+        {
+            private Mobile m_From;
+            private CraftSystem m_CraftSystem;
+            private BaseTool m_Tool;
+            private BaseHat m_Hat;
+
+            public InternalTarget2(Mobile from, CraftSystem craftSystem, BaseTool tool, BaseHat hat) : base(2, false,
+                TargetFlags.None)
+            {
+                m_From = from;
+                m_CraftSystem = craftSystem;
+                m_Tool = tool;
+                m_Hat = hat;
+            }
+
+            private static bool IsHelm(object targeted)
+            {
+                return targeted is Bascinet || targeted is CloseHelm || targeted is Helmet || targeted is NorseHelm ||
+                       targeted is PlateHelm || targeted is ChainCoif || targeted is BoneHelm || targeted is LeatherCap;
+            }
+
+            protected override void OnTarget(Mobile from, object targeted)
+            {
+                int num = m_CraftSystem.CanCraft(from, m_Tool, null);
+
+                if (num > 0)
+                {
+                    from.SendGump(new CraftGump(from, m_CraftSystem, m_Tool, num));
+                    return;
+                }
+
+                if (targeted is BaseArmor helm && IsHelm(helm))
```

Comment:
Might be better to check the BodyPosition of the item? e.g. `helm.BodyPosition == ArmorBodyType.Helmet`. That way you don't need to maintain a list of valid obj types. 

You might also want to turn this into a more abstract system that's not tied to hats but to any armour that implements say `IFortifiable`